cs251tk/specs/util.py

```diff
@@ -4,7 +4,15 @@
 
 def get_filenames(spec):
     """returns the list of files from an assignment spec"""
-    return [file['filename'] for file in spec['files']]
+    files = [file['filename'] for file in spec['files']]
```

Comment:
`return [file['filename'] for file in spec['files'] if not file[options].get('optional', False)]`

Reply:
The problem with that is that it doesn't catch KeyErrors, which are almost guaranteed for required files

Reply:
Nevermind, that does work. `options` should be in quotes, that's where the error was coming from.